### ISSUE TITLE
first start of container fails?

### ISSUE THREAD

**Issue by feliksik** [rancherio/rancher#178]
The first time I start a new container, it seems to fail often. Also if the network container is already created. Then I destroy the container, start again, and it works. Maybe because the pull&start in one go doesn't  work properly. 

Am I the only one experiencing this?

**Comment by ibuildthecloud** [comment]
@feliksik from looking at your logs it's seems the network agent is having difficulty starting on your system.  Those ECONNREFUSED errors are related to the network agent container not working.

We attempted to fix issue around this but ran into a severe bug in docker https://github.com/docker/docker/issues/11087

We have a work around now but it's not fully tested.  What I supposed is happening is that on startup of the network agent it downloads some code and that is hanging.  The fix we had before removed the need to download this code, but we had to roll it back due to a bug in Docker (well, to be honest the bug is in golang itself)
